llama-index-integrations/llms/llama-index-llms-azure-inference/tests/test_llms_azure_inference.py

```diff
@@ -118,3 +121,28 @@ def echo(message: str) -> str:
         response.message.additional_kwargs["tool_calls"][0]["function"]["name"]
         == "echo"
     )
+
+
+@pytest.mark.skipif(
+    not {
+        "AZURE_INFERENCE_ENDPOINT",
+        "AZURE_INFERENCE_CREDENTIAL",
+    }.issubset(set(os.environ)),
+    reason="Azure AI endpoint and/or credential are not set.",
+)
+def test_get_metadata(caplog):
```

Comment:
I think we can use `with pytest.raises(Exception):` if we want to test when something should raise an exception like in this case.

Reply:
Actually, this case was made to avoid making the code failing if the endpoint doesn't support metadata retrieval. There are some operations in `llama-index` that seems to call the metadata property. We would like to prevent such code to fail. Does it make sense? The exception should be catch and a warning logged. Does it make sense?

Reply:
ohhh my fault, I misread the assertion statement! All good, thank you.